apis/apps/v1alpha1/nebulacluster_types.go

```diff
@@ -219,6 +221,11 @@ type ExporterSpec struct {
 	MaxRequests int32 `json:"maxRequests,omitempty"`
 }
 
+// AgentSpec defines the desired state of Agent
+type AgentSpec struct {
```

Comment:
There is no need to embed PodSpec here as it is only used for a component. Instead, you can specify the necessary fields within the AgentSpec, which are three in total.
Image
Version
Resources

By the way, I think renaming AgentSpec to AgentContainerSpec would be better.

Reply:
Thank you for your suggestion. I have revised it